src/main/java/Deadline.java

```diff
@@ -0,0 +1,52 @@
+public class Deadline extends Task{
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public static String getDesc(char[] input) {
+        int marker = 0;
+        String desc = "";
+        for (int i = input.length - 1; (input[i] != '/'); i--) {
+            marker = i;
+        }
+        for (int i = 9; i < marker - 2; i++) {
+            desc += input[i];
+        }
+        return desc;
```

Comment:
Consider adding an exception to be thrown when doing input validation. This way you can just do input.split("/by") to get the deadline date out. If there is a wrong input, you are able to handle that exception as well instead of getting an exception that is not handled.